libs/core/type_support/include/hpx/type_support/generator.hpp

```diff
@@ -324,8 +324,13 @@ namespace hpx {
 
                 dealloc_fn const dealloc = [](void* const p,
                                                std::size_t const s) {
+#if defined(HPX_WITH_CXX14_DELETE_OPERATOR_WITH_SIZE)
                     ::operator delete[](
                         p, s + sizeof(std::size_t) + sizeof(dealloc_fn));
+#else
+                    ::operator delete[](p);
+                    (void) s;
+#endif
```

Comment:
```suggestion
                dealloc_fn const dealloc = [](void* const p,
                    [[maybe_unused]] std::size_t const s) {
#if defined(HPX_WITH_CXX14_DELETE_OPERATOR_WITH_SIZE)
                    ::operator delete[](
                        p, s + sizeof(std::size_t) + sizeof(dealloc_fn));
#else
                    ::operator delete[](p);
#endif
```

Reply:
@hkaiser maybe_unused is since c++17, what do you suggest?

Reply:
Ohkay, it'll work out because of the previous checks. Thanks!